[Schema] OSquery optional TLK

gavinelder opened the pull request:
to:
cc: @airbnb/streamalert-maintainers
related to:
resolves:

## Background

The following makes `log_type` an optional top-level key as this is only added by the [TLS logger](https://github.com/osquery/osquery/blob/d373d042606acda3cb46d6424cfe09be6939b554/plugins/logger/tls_logger.cpp#L100) This fixes classification for Filebeat and other non-TLS sources such as Kolide Fleet.  



## Changes

Add `log_type` an optional top-level key to the OSquery Schema.

## Testing

Steps for how this change was tested and verified

chunyong-lin commented on conf/schemas/osquery.json:
```diff
@@ -18,6 +18,7 @@
       "optional_top_level_keys": [
         "decorations",
         "epoch",
+        "log_type",
```
Is it `log_type` also added it `osquery:batch`?

gavinelder replied:
Good question let me double-check , I don't use batch myself, so it may need included in the JSON

gavinelder commented:
@chunyong-lin All good now log_type is also added to batch, when using the tls logger but not part of the standard schema. I have added this to the batch schema.
